File: Python3/TwoSum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:

        m = {}

        #for i in range(0,len(nums)):
        #    diff = target - nums[i]
        #    if diff in m:
        #        return [m[diff],i]
        #    else:
        #        m[nums[i]] = i

        for i, v in enumerate(nums):
            diff = target - v
            if diff in m:
                return [m[diff],i]
            else:
                m[v] = i

File: Golang/Two Sum.go
func twoSum(nums []int, target int) []int {

    m := make(map[int]int)

    for i := 0; i < len(nums); i++ {
        diff := target - nums[i]

        if _, ok := m[diff]; ok {
            return []int{m[diff],i}
        } else {
            m[nums[i]] = i
        }
    }
    
    return nil
}